src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+import {
+  makeFollower,
+  makeLeader,
+} from '@agoric/casting';
+import { iterateReverse } from '@agoric/casting';
+import { getAllJobs, createDBs, createJob, deleteJob, queryTable, updateTable} from '../helpers/db.js'
+
+// get environment variables
+const {
+  PORT = '3000',
+  EI_CHAINLINKURL,
+  FROM,
+  SUBMIT_RETRIES = 3,
+  BLOCK_INTERVAL = '6',
+  SEND_CHECK_INTERVAL = '45',
+  AGORIC_RPC = "http://0.0.0.0:26657",
+  STATE_FILE = "data/middleware_state.json",
+  CREDENTIALS_FILE = "config/ei_credentials.json",
+  OFFERS_FILE = "config/offers.json",
+  FEEDS_FILE = "oracle/feeds.json"
+} = process.env;
+
+/** 
+  * Environment variables validation
+  */
+if (process.env.NODE_ENV != "test"){
+  assert(EI_CHAINLINKURL, '$EI_CHAINLINKURL is required');
+  assert(Number(SUBMIT_RETRIES), '$SUBMIT_RETRIES is required');
+  assert(FROM, '$FROM is required');
+  assert(validUrl(AGORIC_RPC), '$AGORIC_RPC is required');
+  assert(STATE_FILE != "", '$STATE_FILE is required');
+  assert(CREDENTIALS_FILE != "", '$CREDENTIALS_FILE is required');
+  assert(OFFERS_FILE != "", '$OFFERS_FILE is required');
+  assert(FEEDS_FILE != "", '$FEEDS_FILE is required');
+}
+
+var agoricNames = {};
+var fromBoard = {};
+var vstorage = {};
+const marshaller = boardSlottingMarshaller();
+
+var credentials;
+var feeds;
+
+/**
+  * Function to initialise state
+  */
+const initialiseState = async () => {
+  //create tables if they do not exist
+  await createDBs();
+}
+
+/**
+  * Function to send a job run to the CL node
+  * @param {*} credentials the external initiator credentials
+  * @param {*} count the request id
+  * @param {*} jobId the Chainlink external job id
+  * @param {*} chainlinkUrl the Chainlink node url where to send the job request
+  * @param {*} requestType the request type, 1 = time, 2 = deviation, 3 = new round
+  */
+const sendJobRun = async (credentials, count, jobId, chainlinkUrl, requestType) => {
+  const options = {
+    url: chainlinkUrl + "/v2/jobs/" + jobId + "/runs",
+    body: {
+      "payment": 0,
+      "request_id": count,
+      "request_type": requestType
+    },
+    headers: {
+      "Content-Type": "application/json",
+      "X-Chainlink-EA-AccessKey": credentials["EI_IC_ACCESSKEY"],
+      "X-Chainlink-EA-Secret": credentials["EI_IC_SECRET"]
+    },
+    method: 'POST',
+  };
+
+  //try request with loop retries
+  for (let i = 0; i < SUBMIT_RETRIES; i++) {
+    try {
+      await axios.post(options.url, options.body, {
+        timeout: 60000,
+        proxy: false,
+        headers: options.headers,
+        httpAgent: new http.Agent({ keepAlive: false })
+      });
+      return
+    }
+    catch (err) {
+      console.error("JOB Request for " + jobId + " failed", err)
+    }
+  }
+}
+
+/**
+  * Function to get last 10 offers
+  * @param {*} follower offers and balances
+  * @returns a list of offers
+  */
+export const getOffers = async (follower) => {
```

Comment:
if you're going to return the latest 10 offers say it in the function name. getOffers implies all offers. 

getLatest10Offers for e.g.

Reply:
I think it is okay to stay this way, otherwise, if we decide to change it from 5 to another number we would also have to change the function name which is not scalable. As long as the name is not getAllOffers, it should be ok.